### ISSUE TITLE
Monsters extra turn

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#176]
```
Copied from the forums. User ScorpionZ submitted the following:

In the course of a normal fighting sequence, regardless of who starts the 
fight, it goes something like this:

My turn - three attacks
Monster turn - 1, 2, 3, or 4 (I think the toothy monster gets 4 attacks. Can't 
remember, haven't been in that hole much)
There is a 0.5 to 1.0 second pause then my turn starts again. I believe the 
game is running calculations, but I am not sure. Regardless, there is always a 
pause after the monster's turn.

Every now and then, I am too quick to hit the inventory icon and I catch my 
self having been hit twice (for gargoyle trainers) but unable to apply a 
healing item because it is not my turn yet. When I back out, without fail, the 
monster gets another turn.

Effectively, if I am accidentally too fast in trying to heal myself, I get 
punished by getting my turn skipped. Most of the time that this occurs, it 
results in my death.

Likewise, I have had senarios where I have moved to a tile and a monster moves 
to an adjacent tile and attacks me. Immediately, I see a red number pop up or a 
red splash indicating that I have been hit. No text is displayed. Then the 
monster takes its actual turn showing red splashes and texts for each hit or 
miss. Effectively, there is a bug that allows monsters to get an additional hit 
when they initiate an attack while you are moving.

These bugs don't occur all the time. I don't always remember the one's where 
the monster gets an extra initial hit, but its hard to forget when you get 
skipped a turn and lose 10,000 xp because of it.

Thanks!

Source: http://andors.techby2guys.com/viewtopic.php?f=3&t=140


```

Original issue reported on code.google.com by `sdeva...@gmail.com` on 6 Apr 2011 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
Possible duplicate of issue 132.
```

Original comment by `oskar.wi...@gmail.com` on 7 Apr 2011 at 9:48

**Comment by GoogleCodeExporter** [comment]
```
I have also experienced the move bug before, I think, but I'm not sure.
```

Original comment by `jch...@email.wm.edu` on 8 Apr 2011 at 4:37

**Comment by GoogleCodeExporter** [comment]
```
They are the same, you can merge these if you like.
```

Original comment by `sdeva...@gmail.com` on 9 Apr 2011 at 7:44

**Comment by GoogleCodeExporter** [comment]
```
when the monster is attacking you can skip the attack phase by tapping the 
screen or some other button. It's just something done so that way you can see 
each hit or miss the monster makes.
```

Original comment by `dope...@gmail.com` on 27 Apr 2011 at 2:41

**Comment by GoogleCodeExporter** [comment]
```
As of v0.6.10, the game supports saving monster's current AP, which means that 
this is theoretically solvable now.

With the bugfixes done in r237, this should now be fully resolved.
```

Original comment by `oskar.wi...@gmail.com` on 29 Apr 2012 at 9:49
- Changed state: **Ready**
- Added labels: **Milestone-0.6.11**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.11. Closing issue.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jun 2012 at 9:31
- Changed state: **Fixed**
